File: Leetcode/78.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        constexpr int MAXN = 12;
        bitset<MAXN> vis;
        vector<vector<int>> answer;
        int n = (int)nums.size();
        int allPossibles = 1 << n;
        for (int i = 0; i < allPossibles; ++i) {
            vector<int> result;
            for (int j = 0; j < n; ++j) {
                if (vis[j]) {
                    result.emplace_back(nums[j]);
                }
            }
            answer.emplace_back(result);
            vis = vis.to_ulong() + 1;
        }
        return answer;
    }
};

File: Leetcode/78/78.go
func g(n int, nums []int) []int {
	var s []int
	for n > 0 {
		if (n & 1) != 0 {
			s = append(s, 1)
		} else {
			s = append(s, 0)
		}
		n >>= 1
	}
	for len(s) < len(nums) {
		s = append(s, 0)
	}
	var res []int
	for i := 0; i < len(nums); i++ {
		if s[i] == 1 {
			res = append(res, nums[i])
		}
	}
	return res
}
func subsets(nums []int) [][]int {
	n := len(nums)
	tot := 1 << n
	var res [][]int
	for i := 0; i < tot; i++ {
		res = append(res, g(i, nums))
	}
	return res
}